Makefile

```diff
@@ -39,18 +39,18 @@ test:
 .PHONY: external-mypy
 external-mypy:
 	@echo "testing simple example with mypy (and python to check it's sane)..."
-	python tests/mypy_test_success.py
-	mypy tests/mypy_test_success.py
+	python tests/mypy/success.py
```

Comment:
can this now be removed?

Reply:
I think it is safe to remove `external-mypy` entirely, but given that `mypy.api.run` only works (currently) if `typing_extensions` is installed, I wasn't sure whether you'd be okay removing this as it would disable mypy testing when `typing_extensions` is not installed.

At the very least, we can probably remove the `python tests/mypy/success.py` since that should be run under all circumstances. (Maybe that's specifically what you meant?)

Reply:
I meant removing it completely.

Why do mypy tests require typing-extensions? Could we split the tests so only some require typing-extensions?

Reply:
`mypy.api.run` tries to import typing_extensions; it caused the build to fail until I added the `skipif`s to the tests. I'm not sure if that's because it is seeing `typing_extensions.Literal` in the code, or if it always tries to import typing_extensions. I can dig into this more this week if desired.

Reply:
Ye, I think that's the best solution, we'll probably end up with two different tests (or sets of tests):
* one that uses type-extensions and will need to be run with `skipIf`
* and the other that doesn't

Reply:
@samuelcolvin I looked into this more:

* As far as I can tell, `typing_extensions` is actually a **dependency** of `mypy`
  * I'm assuming the build server is running mypy from a separate environment? Otherwise typing_extensions should be installed. (Or maybe it is only uninstalled because we added explicit uninstallation?)
* Because of this, I don't think it's possible for mypy to be used without typing_extensions installed.

Thus, I think we are safe to remove `external-mypy`, and also keep the `skipif`s on the tests for when `typing_extensions` is not installed. Let me know what you think.

Reply:
sounds good to me.

As per:

https://github.com/samuelcolvin/pydantic/blob/321cde0c8888642005986e109c2643efba96eb9a/.travis.yml#L57-L64

The "Without Deps" tests uninstall `typing-extensions` before running `make test`.